### PR TITLE
React navigation fix to prevent multiple navigations when pressing button

### LINKED ISSUE #42: Implement this Fix for prevent navigating twice o more times when clicking a button quickly
In store.js ...
```
const getCurrentRouteName = state => {
  const route = state.routes[state.index];
  return typeof route.index === 'undefined' ? route.routeName : getCurrentRouteName(route);
};

const nav = (state, action) => {
  const nextState = Navigator.router.getStateForAction(action, state);

  // prevents navigating twice to the same route
  if (state && nextState) {
    const stateRouteName = getCurrentRouteName(state);
    const nextStateRouteName = getCurrentRouteName(nextState);
    return stateRouteName === nextStateRouteName ? state : nextState;
  }

  // Simply return the original `state` if `nextState` is null or undefined.
  return nextState || state;
};
```

### PR DESCRIPTION
**Pull request by WFolini**
### Summary
Implement the fix that solves #42

### Screenshots
No visual changes.

### Trello Card
https://trello.com/c/6xpUiBmw/61-prevent-multiple-navigations-when-pressing-button

### CODE REVIEW

File: generators/app/templates/src/utils/navUtils.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export const getCurrentRouteName = navigationState => {
```
Comment (mvbattan) [inline generators/app/templates/src/utils/navUtils.js]: Isn't this part of another PR ? 🤔

File: generators/app/templates/src/redux/store.ejs [hunk 2]
```diff
@@ -16,7 +18,28 @@ import drawer from './drawer/reducer';
...
+    const nextStateRouteName = getCurrentRouteName(nextState);
+    // Check if has same params
+    const hasSameParams = isEqual(currentParams, nextParams);
+    // Check if has same route
+    const hasSameRoute = stateRouteName === nextStateRouteName;
+    // Return original state if have same routes with same params and next state in other case
+    if (hasSameRoute) {
+      return hasSameParams ? state : nextState;
+    }
+    return nextState;
```
Comment (mvbattan) [inline generators/app/templates/src/redux/store.ejs]: This line is redundant

File: generators/app/templates/src/redux/store.ejs [hunk 3]
```diff
@@ -16,7 +18,28 @@ import drawer from './drawer/reducer';
...
+  const nextState = Navigator.router.getStateForAction(action, state);
+  // Prevents navigating twice to the same route but allow navigate if same routes with different params
+  if (state && nextState) {
+    // Getting routes and params
+    const currentParams = Navigator.router.getPathAndParamsForState(state).params;
+    const nextParams = action.params;
+    const stateRouteName = getCurrentRouteName(state);
+    const nextStateRouteName = getCurrentRouteName(nextState);
+    // Check if has same params
+    const hasSameParams = isEqual(currentParams, nextParams);
```
Comment (mvbattan) [inline generators/app/templates/src/redux/store.ejs]: I'd check for `hasSameParams` inside the `if (hasSameRoute)` block so we avoid doing `isEqual` always.

File: generators/app/templates/src/utils/navUtils.js (hunk 1 above)
Comment (WFolini) [reply generators/app/templates/src/utils/navUtils.js]: Mmmm yes, but it's needed in this PR also. When the one who merged first, the other will update and these changes will disappear

**Comment by WFolini** [comment]
@mvbattan Done!

**Pull request merged**